core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -328,11 +338,23 @@ private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nu
 
       if (!excelInputreaderConfig.containsMacro("memoryTableName") &&
         !Strings.isNullOrEmpty(excelInputreaderConfig.memoryTableName)) {
+
+        ExploreProperties.Builder builder = ExploreProperties.builder();
+        if (!Strings.isNullOrEmpty(excelInputreaderConfig.memoryDatasetExplorer)) {
```

Comment:
Value for 'memoryDatasetExplorer' is based on value of 'memoryTableName' and there should be no default value for 'memoryTableName', that's the reason 'memoryDatasetExplorer' could not be set in the constructor.